frontend/src/views/ProjectView.vue

```diff
@@ -1,29 +1,33 @@
 <script setup lang="ts">
+import { format } from "date-fns";
 import Simplebar from "simplebar-vue";
 import { onBeforeUnmount, onMounted, ref } from "vue";
 
-import ReportCanvas from "@/components/ReportCanvas.vue";
+import EditableList, { type EditableListItemModel } from "@/components/EditableList.vue";
+import ProjectViewCanvas from "@/components/ProjectViewCanvas.vue";
 import SectionHeader from "@/components/SectionHeader.vue";
 import SimpleButton from "@/components/SimpleButton.vue";
 import TreeAccordion from "@/components/TreeAccordion.vue";
 import { fetchShareableBlob } from "@/services/api";
-import { saveBlob } from "@/services/utils";
-import { useReportStore } from "@/stores/report";
+import { generateRandomId, saveBlob } from "@/services/utils";
+import { useProjectStore } from "@/stores/project";
+import { useToastsStore } from "@/stores/toasts";
 
-const reportStore = useReportStore();
+const projectStore = useProjectStore();
+const toastsStore = useToastsStore();
 const isDropIndicatorVisible = ref(false);
 const editor = ref<HTMLDivElement>();
 const isInFocusMode = ref(false);
+const views = ref<EditableListItemModel[]>([]);
+let unsavedViewsIds: string[] = [];
```

Comment:
question: Is this used to manage delayed view saving? I am not sure to know why a view would not be saved.

Reply:
It is used to differentiate views who are not know by the backend. It's a mapping of pending views. For example you create a view and did not gave it a name then it's local id is in this array and the project view knows that it's a creation and not a rename.